<?xml version="1.0"?>
<robot xmlns:xacro="http://www.ros.org/wiki/xacro">

    <!-- ros2_control intiliazisation -->
    <ros2_control name="GazeboSimSystem" type="system">
        <hardware>
            <plugin>gazebo_ros2_control/GazeboSystem</plugin>
        </hardware>
        <joint name="base_yaw_joint">
            <command_interface name="effort">
                <param name="min">-1000000</param>
                <param name="max">1000000</param>
            </command_interface>
            <state_interface name="position"/>
            <state_interface name="velocity"/>
            <state_interface name="effort"/>
        </joint>
        <joint name="yaw_bicep_joint">
            <command_interface name="effort">
                <param name="min">-1000000</param>
                <param name="max">1000000</param>
            </command_interface>
            <state_interface name="position"/>
            <state_interface name="velocity"/>
            <state_interface name="effort"/>
        </joint>
        <joint name="bicep_forearm_joint">
            <command_interface name="effort">
                <param name="min">-1000000</param>
                <param name="max">1000000</param>
            </command_interface>
            <state_interface name="position"/>
            <state_interface name="velocity"/>
            <state_interface name="effort"/>
        </joint>
        <joint name="forearm_endeffector_joint">
            <command_interface name="effort">
                <param name="min">-1000000</param>
                <param name="max">1000000</param>
            </command_interface>
            <state_interface name="position"/>
            <state_interface name="velocity"/>
            <state_interface name="effort"/>
        </joint>

        <!-- Grippe joints -->
        <joint name="a1_joint">
            <command_interface name="effort">
                <param name="min">-1000000</param>
                <param name="max">1000000</param>
            </command_interface>
            <state_interface name="position"/>
            <state_interface name="velocity"/>
            <state_interface name="effort"/>
        </joint>
        <joint name="a2_joint">
            <command_interface name="effort">
                <param name="min">-1000000</param>
                <param name="max">1000000</param>
            </command_interface>
            <state_interface name="position"/>
            <state_interface name="velocity"/>
            <state_interface name="effort"/>
        </joint>
        <joint name="b1_joint">
            <command_interface name="effort">
                <param name="min">-1000000</param>
                <param name="max">1000000</param>
            </command_interface>
            <state_interface name="position"/>
            <state_interface name="velocity"/>
            <state_interface name="effort"/>
        </joint>
        <joint name="b2_joint">
            <command_interface name="effort">
                <param name="min">-1000000000</param>
                <param name="max">1000000</param>
            </command_interface>
            <state_interface name="position"/>
            <state_interface name="velocity"/>
            <state_interface name="effort"/>
        </joint>
    </ros2_control>

    <gazebo>
        <plugin name="gazebo_ros2_control"
            filename="libgazebo_ros2_control.so">
            <parameters>$(find qarm_v2)/config/controllers.yaml</parameters>
        </plugin>
    </gazebo>

</robot>